src/scheduler/check.c

```diff
@@ -1341,10 +1341,9 @@ find_counts_elm(counts *cts_list, char *name, resdef *rdef, counts **cnt, resour
 			*cnt = cts;
 		if (rdef == NULL)
 			return cts->running;
-		else {
-			if ((res_lim = find_resource_count(cts->rescts, rdef)) != NULL)
-				if (rcount != NULL)
-					*rcount = res_lim;
+		else if ((res_lim = find_resource_count(cts->rescts, rdef)) != NULL) {
+			if (rcount != NULL)
+				*rcount = res_lim;
 			return res_lim->amount;
```

Comment:
So, I think this might be changing existing logic. Previously, the code would return res_lim->amount regardless of the if ((res_lim = find_resource_count(cts->rescts, rdef)) condition, now it will return it only if that condition is true, is this change intentional?

Reply:
Right that's the bug, because, find_resource_count returns "res_lim". so unless it is not-null, we end up dereferencing an uninitialized pointer.